@AH-64D Apache Longbow/Addons/fza_ah64_dms/functions/route/fn_routeData.sqf

```diff
@@ -0,0 +1,111 @@
+/* ----------------------------------------------------------------------------
+Function: fza_dms_fnc_routeData
+
+Description:
+    Handles the cycling and data proccessing for routes
+
+Parameters:
+    _heli - The apache
+
+Returns:
+    Nothing
+
+Examples:
+    --- Code
+    [_heli] call fza_dms_fnc_routeData
+    ---
+
+Author:
+    Snow(Dryden)
+---------------------------------------------------------------------------- */
+#include "\fza_ah64_dms\headers\constants.h"
+params ["_heli"];
+
+if !(local _heli) exitWith {};
+
+private _routeData    = _heli getVariable "fza_ah64_routeData";
+private _routeCurrent = _heli getVariable "fza_ah64_routeSelected";
+private _routeInfo    = _routeData # _routeCurrent;
+private _rteIndex     = _heli getVariable "fza_ah64_routeCurPoint";
+private _Course1      = -1;
+private _Course2      = -1;
+
+
+//create Route without A/B/B/C side/side duplicates
+private _rteCycleList = [];
+private _rteDrawList = [];
+private _rteCycleIndex = _rteIndex;
+{
+    private _previousIndex = (_foreachindex - 1);
+    private _previousPoint = (_routeInfo#_previousIndex);
+    if (_previousPoint isEqualTo _x && _previousIndex > -1) then { 
+        if (_previousIndex < _rteIndex) then {
+            _rteCycleIndex = _rteCycleIndex - 1;
+        };
+        continue;
+    };
+    _rteCycleList pushBack _X;
+    _rteDrawList pushback ([_heli, _X, POINT_GET_ARMA_POS] call fza_dms_fnc_pointGetValue);
+} foreach _routeInfo;
+
+[_heli, "fza_dms_routeDrawArray", _rteDrawList] call fza_fnc_updateNetworkGlobal;
```

Comment:
I do not see why `fza_dms_routeDrawArray` actually exists. We can directly create this in `pageTSDDrawCanvas` from the same variables

Reply:
In there rather than having the `_drawArray`, just read the current route

Reply:
done